src/open_clip/hf_model.py

```diff
@@ -132,12 +131,21 @@ def __init__(
                 nn.Linear(hidden_size, output_dim, bias=False),
             )
 
-    def forward(self, x: TensorType) -> TensorType:
+    def forward(self, x: TensorType, output_tokens=False) -> TensorType:
         attn_mask = (x != self.config.pad_token_id).long()
         out = self.transformer(input_ids=x, attention_mask=attn_mask)
         pooled_out = self.pooler(out, attn_mask)
+        projected = self.proj(pooled_out)
 
-        return self.proj(pooled_out)
+        if output_tokens:
+            tokens = self.proj(
+                out.last_hidden_state[:, 1:, :] 
```

Comment:
might make code a bit ugly but at least let's save ourselves some really confusing bugs in the form of just not taking the CLS token

Reply:
current version doesn't fix it either. That's why I said it's going to be a little ugly because we need to do tokens = everything but the CLS token at the given position. I'm fine with only supporting the 2 most common options: first position and last position

Reply:
so maybe instead of slicing you can do something like tok = all_tokens and then drop cls_token_position

Reply:
```
        if output_tokens:
            seq_len = out.last_hidden_state.shape[1]
            tokens = self.proj(
                out.last_hidden_state[:, torch.arange(seq_len) != self.pooler.cls_token_position, :] 
                if type(self.pooler) == ClsPooler 
                else out.last_hidden_state
            )
            return projected, tokens
```

Reply:
is this ok?

Reply:
I think so, if it works ofc.